Formula/d/dub.rb

```diff
@@ -8,8 +8,8 @@ class Dub < Formula
   head "https://github.com/dlang/dub.git", branch: "master"
 
   livecheck do
-    url :stable
-    regex(/^v?(\d+(?:\.\d+)+)$/i)
+    url "https://code.dlang.org/packages/dub"
+    regex(%r{"badge">v?(\d+(?:\.\d+)+)</strong>}i)
   end
```

Comment:
The dub registry actually seems to provide an API for retrieving the latest version of a package: https://code.dlang.org/api/packages/dub/latest (it's undocumented; I discovered it [in their codebase](https://github.com/dlang/dub-registry/blob/ce84086011862249d64b21fda8c31a14e0935ee7/source/dubregistry/api.d#L83-L84)).

It returns a JSON-quoted string:

```console
$ curl https://code.dlang.org/api/packages/dub/latest
"1.35.0"
```

However, I am not sure if the `Json` strategy is the best fit. The following works; while the block seems redundant, I could not make it work without that:

```ruby
  livecheck do
    url "https://code.dlang.org/api/packages/dub/latest"
    strategy :json do |json|
      json
    end
  end
```

Any thoughts, @samford?

Reply:
Thanks for the explanation! Updating livecheck at #156443.

> However, that would complicate the mental model of how the strategy works and I'm not entirely convinced it would be worth it (i.e., uniformity is more valuable to me, even if it sometimes requires a minimal `strategy` block like this).

Agreed. 👍

Reply:
actually now the registry returns beta release

```
$ curl https://code.dlang.org/api/packages/dub/latest
"1.36.0-beta.1"
```

Reply:
I just went ahead to bump to the latest stable release, https://github.com/Homebrew/homebrew-core/pull/160245

Reply:
`https://code.dlang.org/api/packages/dub/info?minimize=true ` is essentially the github tag list.

Reply:
It depends if we need to check that the package exists on the registry, and not just on github.